gui/qt/main_window.py

```diff
@@ -2248,7 +2248,7 @@ def do_encrypt(self, message_e, pubkey_e, encrypted_e):
             encrypted_e.setText(encrypted.decode('ascii'))
```

Comment:
I think only the first line (`public_key = ...`) actually needs to be protected against `BaseException`. Given you are now showing a warning that specifically says the public key is invalid, let's move the other two lines out of the try, and `return` in the `except` branch after showing the warning (or use an `else` branch).

Reply:
Yes. Altered the code.